agimus_controller/agimus_controller/trajectories/sine_wave_cartesian_space.py

```diff
@@ -29,24 +32,187 @@ def __init__(
             scale_duration=sine_wave_params.scale_duration
         )
         self.amp = sine_wave_params.amplitude
-        self.w = 2.0 * np.pi / sine_wave_params.period  # pulsation
+        self.w = sine_wave_params.pulsation
         self.w_q = w_q
         self.w_qdot = w_qdot
         self.w_qddot = w_qddot
         self.w_robot_effort = w_robot_effort
         self.w_pose = w_pose
+        self.optimize_orientation = True
+
+        # Desired end-effector trajectory parameters.
+        self.ee_init_pos = pin.SE3.Identity()
+        self.ee_des_pos = pin.SE3.Identity()
+        self.ee_des_vel = np.zeros(6)
+
+        # Inverse kinematics parameters.
+        self.ik_error = np.zeros(6)
+        self.ik_precision = 1e-4
+        self.ik_it_max = 100
+        self.ik_dt = 1e-1
+        self.ik_damp = 1e-12
+        self.ik_p_error = np.zeros(3)
+        self.ik_ee_pose = pin.SE3.Identity()
+        self.ik_success = False
+        self.ik_q = None
+        self.ik_dq = None
+        self.ik_Jp = None
+        self.ik_Jdamp = None
+
+        self.ik3_precision = 1e-3
+        self.ik3_it_max = 10000
+        self.ik3_dt = 1e-3
+        self.ik3_damp = 1e-12
+        self.ik3_error = np.zeros(3)
+        self.ik3_Jp = None
+        self.ik3_Jdamp = None
+
+    def initialize(self, pin_model: pin.Model, q0: np.ndarray) -> None:
+        """Initialize the trajectory generator."""
+        super().initialize(pin_model, q0)
+        self.ik_q = self.q0.copy()
+        self.ik_dq = np.zeros(self.pin_model.nv)
+        self.ik_Jp = np.zeros((6, self.pin_model.nv))
+        self.ik_Jdamp = np.eye(6)
+        self.ik3_Jp = np.zeros((3, self.pin_model.nv))
+        self.ik3_Jdamp = np.eye(3)
+        self.ee_init_pos = self.get_end_effector_pose_from_q_as_se3(self.q0)
```

Comment:
I guess you initialize the memory for computation time reasons. At the best, the computation time is not affected. At worst, it is increased.

Moreover, it makes all lines in the IK algo longer and less readable because of all the `self.`
I suggest to remove those attribute and keep them as variables internal to the function.
I don't think you need to initialize the memory.